ui/app/AppLayouts/Communities/views/CommunityColumnView.qml

```diff
@@ -95,6 +95,14 @@ Item {
                 }
             }
         }
+
+        Connections {
+            target: communitySectionModule
+            function onUserAuthenticationCanceled() {
+                joinCommunityButton.invitationPending = false
+                joinCommunityButton.loading = false
+            }
+        }
```

Comment:
Nice

Could you please do the same in Settings/Communities (in CommunitiesListPanel.qml), there is:

```qml
readonly property bool isInvitationPending: root.rootStore.isCommunityRequestPending(model.id)
```
which needs similar treatment :)

`root.rootStore.isCommunityRequestPending(model.id)` should really be a model entity, then we wouldn't need these kind of repeated hacks above :/

We have roles like `model.joined`, so why not `model.invitationPending` for the given community?

Reply:
I just tested it out and while it does have an issue of not updating correctly and it would be better using a model, it's not suffering from the issue above, so I'll just create another issue for that one.